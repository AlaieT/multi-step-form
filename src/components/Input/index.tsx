import React from "react";

import classNames from "../../utils/classNames";

import type { InputProps } from "../../types";

import styles from "../../styles/components/input.module.scss";

const Input = React.forwardRef<HTMLInputElement, InputProps>(
  ({
    label, error, className, id, ...rest
  }, ref) => (
    <div className={classNames(styles.input, className)}>
      <div className={styles.badges}>
        <label htmlFor={id}>{label}</label>
        {error && <label htmlFor={id}>{error}</label>}
      </div>
      <input {...rest} id={id} ref={ref} className={error && styles.error} />
    </div>
  )
);

Input.displayName = "Input";

export default Input;
