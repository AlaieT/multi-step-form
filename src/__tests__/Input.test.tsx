import React from "react";
import { render, screen, fireEvent, cleanup } from "@testing-library/react";

import Input from "../components/Input";

afterEach(cleanup);

describe("Input", () => {
  describe("correct render", () => {
    it("should match snapshot", () => {
      const { asFragment } = render(
        <Input id="name" label="Name" error="Name is required!" />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("correct functionality", () => {
    it("should does correct input", () => {
      render(<Input id="name" label="Name" />);
      fireEvent.input(screen.getByRole("textbox", { name: "Name" }), {
        target: { value: "Some name" }
      });
      expect(screen.getByRole("textbox", { name: "Name" })).toHaveValue(
        "Some name"
      );
    });
  });
});
