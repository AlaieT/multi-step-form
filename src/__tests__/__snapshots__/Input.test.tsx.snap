// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input correct render should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="input"
  >
    <div
      class="badges"
    >
      <label
        for="name"
      >
        Name
      </label>
      <label
        for="name"
      >
        Name is required!
      </label>
    </div>
    <input
      class="error"
      id="name"
    />
  </div>
</DocumentFragment>
`;
